stm32cube/stm32h7rsxx/CMakeLists.txt

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) 2019  Linaro Limited
+# Copyright (c) 2020 STMicroelectronics
+# Copyright (c) 2024 STMicroelectronics
+#
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_library_sources(soc/system_stm32h7rsxx.c)
+zephyr_library_sources(drivers/src/stm32h7rsxx_hal.c)
+zephyr_library_sources(drivers/src/stm32h7rsxx_hal_rcc.c)
+zephyr_library_sources(drivers/src/stm32h7rsxx_hal_rcc_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ADC drivers/src/stm32h7rsxx_hal_adc.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ADC_EX drivers/src/stm32h7rsxx_hal_adc_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_CEC drivers/src/stm32h7rsxx_hal_cec.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_COMP drivers/src/stm32h7rsxx_hal_comp.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_CORDIC drivers/src/stm32h7rsxx_hal_cordic.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_CORTEX drivers/src/stm32h7rsxx_hal_cortex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_CRC drivers/src/stm32h7rsxx_hal_crc.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_CRC_EX drivers/src/stm32h7rsxx_hal_crc_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_CRYP drivers/src/stm32h7rsxx_hal_cryp.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_CRYP_EX drivers/src/stm32h7rsxx_hal_cryp_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DAC drivers/src/stm32h7rsxx_hal_dac.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DAC_EX drivers/src/stm32h7rsxx_hal_dac_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DCMI drivers/src/stm32h7rsxx_hal_dcmi.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DFSDM drivers/src/stm32h7rsxx_hal_dfsdm.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DFSDM_EX drivers/src/stm32h7rsxx_hal_dfsdm_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DMA drivers/src/stm32h7rsxx_hal_dma.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DMA2D drivers/src/stm32h7rsxx_hal_dma2d.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DMA_EX drivers/src/stm32h7rsxx_hal_dma_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DSI drivers/src/stm32h7rsxx_hal_dsi.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_DTS drivers/src/stm32h7rsxx_hal_dts.c)
+if(CONFIG_ETH_STM32_HAL_API_V2)
+	zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ETH drivers/src/stm32h7rsxx_hal_eth.c)
+	zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ETH_EX drivers/src/stm32h7rsxx_hal_eth_ex.c)
+else()
+	zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ETH drivers/src/Legacy/stm32h7rsxx_hal_eth.c)
+	zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ETH_EX drivers/src/Legacy/stm32h7rsxx_hal_eth_ex.c)
+endif()
```

Comment:
V1 API to be removed. Not available as part of H7RS package and deprecated in Zehyr.